### ISSUE TITLE
Many libraries have really poor error management

### ISSUE THREAD

**Issue by alcuadrado** [ethereumjs/organization#54]
I'm starting to use `-vm`, `-block` and `-blockchain` a lot, and I'm constantly struggling to debug things because strings are passed instead of errors.

This is a good example of a function that is a nightmare to work with: https://github.com/ethereumjs/ethereumjs-block/blob/master/header.js#L207

The simplest way to solve this is to use `new Error("existing string")` instead of `"existing string"` everywhere. Making sure that it's actually fixed **everywhere** is a perfect example of something achievable in a couple of hours in a monorepo, but that would take a lot of time to coordinate across multiple repos.

**Comment by holgerd77** [comment]
Let's just start consolidating, still think one repo to rule them all is too much, but we have suggestions on this and can e.g. set up a first blockchain/block/tx repo or start to integrate account into VM. If we'd end up with 3-5 repos at the end that would already be a huge win.

Otherwise greetings from a mountain top, still on holidays for 10 days. 😋🙂

**Comment by alcuadrado** [comment]
I think the packages that would benefit the most from being in the same repo are vm, block, blockchain, and account. The rest of them are commonly used independently (e.g. tx and util), or rarely changed (e.g. rlp, merkle-patricia-trie).

**Comment by holgerd77** [comment]
Hmmm, I think we should then also take `tx` for consistency (all structure-part-libs in one repo), but otherwise might be a way to do this grouping within the VM repo with the other libs you proposed (so then it would be: vm, blockchain, block, tx, account). @s1na what do you think?

I think this wouldn't overblow yet the issue list and affect manageability too much.

**Comment by s1na** [comment]
Sounds good to me! I've been preaching this for some time now myself :)

> The rest of them are commonly used independently (e.g. tx and util)

Even in a monorepo they'd be publishable to npm. Overally I like the list @holgerd77 made. But maybe we can start consolidating one by one? I'd start with `ethereumjs-account`

**Comment by holgerd77** [comment]
Yes, I think that's a good plan, cool that we have some agreement here now. 😄 I've opened an issue on the VM https://github.com/ethereumjs/ethereumjs-vm/issues/561 and an accompanying issue on the config repo https://github.com/ethereumjs/ethereumjs-config/issues/24 (to update configuration to also work well with a consuming monorepo).
